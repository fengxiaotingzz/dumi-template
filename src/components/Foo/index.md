# Foo

This is an example component.

```jsx
import { Foo } from 'dumi-template';

export default () => <Foo title="Hello dumi!" />;
```
