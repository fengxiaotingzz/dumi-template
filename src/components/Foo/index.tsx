import React, { useEffect, type FC } from 'react';
import { test } from '@/utils';

const Foo: FC<{ title: string }> = () => {
  useEffect(() => {
    test();
  }, []);
  return <div>111</div>;
};

export default Foo;
