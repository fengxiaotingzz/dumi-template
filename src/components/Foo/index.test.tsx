import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import Foo from './index';

describe('组件测试: Button', function () {
  test('render correctly', () => {
    const { asFragment } = render(<Foo title="11" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
