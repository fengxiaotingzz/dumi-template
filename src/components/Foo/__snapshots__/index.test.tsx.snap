// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`组件测试: Button render correctly 1`] = `
<DocumentFragment>
  <div>
    111
  </div>
</DocumentFragment>
`;
